src/main/java/org/thief/service/StealService.java

```diff
@@ -0,0 +1,47 @@
+package org.thief.service;
+
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+import org.thief.dto.NasaPhoto;
+import org.thief.dto.NasaResponse;
+import org.thief.repository.dao.Camera;
+
+import java.net.URI;
+import java.util.Objects;
+
+@Service
+@RequiredArgsConstructor
+public class StealService {
+    private static final String URL_BASE = "https://api.nasa.gov/mars-photos/api/v1/rovers/curiosity/photos";
+    private static final String URL_PARAM_SOL = "sol";
+    private static final String URL_PARAM_API_KEY = "api_key";
+
+    private final RestTemplate restTemplate = new RestTemplate();
+
+    private final CameraService cameraService;
+    private final PictureService pictureService;
+
+    @Value("${thief.nasa-api-key}")
+    private String apiKey;
+
+    @Transactional
+    public void stealNasaData(int sol) {
+        URI requestUri = UriComponentsBuilder.fromUriString(URL_BASE)
+                .queryParam(URL_PARAM_SOL, sol)
+                .queryParam(URL_PARAM_API_KEY, apiKey)
+                .build()
+                .toUri();
+        ResponseEntity<NasaResponse> response = restTemplate.getForEntity(requestUri, NasaResponse.class);
```

Comment:
I recommend to put the executing api logic in separated service e.g. ClientNasaApi

Reply:
I don't see very specific reason to create separate service for that, StealService is the one that does the whole logic and uses other services to manage data. Why do we need at this moment separate service? What benefit it will give us to have it like this?
I think it might be better to put this logic into private method, but not into separate service.

Reply:
So please extract it in private method. You may reuse in another service that also works with Nasa API. But there are no such requirements so far. 

Think how you want to declare transition here. Is it fine for you to rollback if you got an error or do you want to have bunch of transitions, one per each picture?

Reply:
It **may**, so right now I don't need it, it could be easily done when such requirements appear.

Regarding transaction - at this moment it's fine for me to rollback if any error appears.